RFC0018.md

```diff
@@ -0,0 +1,69 @@
+
+## Action
+
+
+## Copyright Notice
+Copyright (c) 2017 Intel, Inc. All rights reserved.
+
+This document is subject to all provisions relating to code contributions to the PMIx community as defined in the community's [LICENSE](https://github.com/pmix/RFCs/tree/master/LICENSE) file. Code Components extracted from this document must include the License text as described in that file.
+
+## Description
+Several use-cases have identified desirable extensions to the behavior of the event notification system, especially in the registration area. These include:
+
+* Hybrid applications (i.e., applications that utilize more than one programming model, such as an MPI application that also uses OpenMP) are growing in popularity. As addressed by the relevant [PR](https://github.com/pmix/RFCs/pull/17), this trend would benefit from cross-library coordination. Coordinating with a threading library such as OpenMP subsequently creates the need for event notifications between threads of the same process, and opens the door for multiple threads to potentially register different event handlers against the same status code.
+
+* a need for the caller to define a desired callback sequence when registering event handlers. This is primarily driven by a desire to ensure that a particular event handler is called prior to a global default event handler that might take draconian action (e.g., abort the job)
+
+* a need for an event handler to indicate whether or not an "abort" is required at completion of the event processing chain. In many cases, the default event handler simply aborts the application. This is adequate in cases where a prior event handler can fully resolve the problem, and therefore safely terminate the event handler callback chain. However, in some cases an event handler may handle only enough of the problem to assure that an "abort" operation is not required, but needs to allow other event handlers an opportunity to further resolve issues. In these cases, an ability to add an attribute to the results that tells follow-on event handlers to limit their response (e.g., to not "abort") is required.
+
+None of these require modification of existing PMIx APIs, nor addition of new ones. Instead, all can be supported by adding attributes to direct the behavior of the existing event registration/notification functions.
+
+#### Event registration extensions
+The primary need here is for attributes indicating desired ordering of the event handler being registered vs other handlers that have already been registered or will subsequently be registered. In both cases, it is necessary that the user provide something to identify each handler so the relative position can later be specified - this is to be done via the existing _PMIX\_EVENT\_HDLR\_NAME_ attribute. In addition, the existing _PMIX\_EVENT\_ORDER\_PREPEND_ attribute directs PMIx to prepend the handler being registered to the front of the chain, as opposed to the default append behavior.
```

Comment:
Shouldn't the default behavior be to prepend? In a typical use case, the application will initialize libraries in a "more generic to more specific" order, meaning that the last registered handler is the one that is the most "specialized". So it most often should have precedence at trying to "fix the problem", before passing the hand to more generic handlers. 

I understand that this additional PR greatly alleviates this concern, but it still seems more logical in that direction to me.